flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -27,9 +27,30 @@ void TableColumn(TableCreationContext context) :
         UniqueKey(context.uniqueKeysList)
     |
         ComputedColumn(context)
+    |
+        Watermark(context)
     )
 }
 
+void Watermark(TableCreationContext context) :
+{
+    SqlIdentifier columnName;
+    SqlParserPos pos;
+    SqlNode watermarkStrategy;
+}
+{
+    <WATERMARK> <FOR>
+    columnName = CompoundIdentifier() {pos = getPos();}
+    <AS>
+    watermarkStrategy = Expression(ExprContext.ACCEPT_SUB_QUERY) {
+        if (context.watermark != null) {
```

Comment:
`ExprContext.ACCEPT_SUB_QUERY` do we really support sub-query ? I don't think so.

Reply:
I referred to the computed column parser. I think maybe both of them should use `ExprContext.ACCEPT_NON_QUERY` ?

Reply:
Yes, the computed column should use `ExprContext.ACCEPT_NON_QUERY` which i have implemented in my logcal changes.